src/airport.js

```diff
@@ -1 +1,59 @@
+class Airport {
+    constructor() {
+        this.airportCapacity = 10;
+        this.airportPlanes = [];
+        this.weatherMapping = {
+            1: 'sunny',
+            2: 'stormy',
+        };
+    }
 
+    landPlane(plane) {
+        if (this.isStormy()) {
+            throw new Error('Cannot land due to stormy weather.');
+        }
+
+        if (this.isFull()) {
+            throw new Error('Airport is full. Cannot land plane.');
+        }
+
+        this.airportPlanes.push(plane);
+    }
+
+    setCapacity(capacity) {
+        this.airportCapacity = capacity;
+    }
+
+    isFull() {
+        return this.airportPlanes.length >= this.airportCapacity;
+    }
+
+    takeOff(plane) {
+        if (this.isStormy()) {
+            throw new Error('Cannot take off due to stormy weather.');
+        }
+
+        if (!this.isInAirport(plane)) {
+            throw new Error('Plane is not in the airport.');
```

Comment:
Good part about the code

It includes error handling for specific scenarios. It throws custom error messages when attempting to land or take off a plane under stormy weather conditions or when the airport is at full capacity. This helps to prevent invalid operations and provides feedback to the user.